### DISCUSSION TITLE
mc `--recursive` not showing all objects, while `mc tree` does

### DISCUSSION THREAD

**Discussion by rdxmb** [minio/minio#20428] [Q&A]
Hello, we have the effect that `mc tree` shows all our objects as expected, but `mc ls --recursive` and `mc mirror` and `mc copy --recursive` show/copy only some of them.

```
sh-5.1# mc tree --files production/short-term-storage | wc -l
26859
sh-5.1# mc ls --recursive production/short-term-storage | wc -l
782
```

We are using a Tenant created by the MinIO Operator.

```
# mc -v
mc version RELEASE.2024-09-09T07-53-10Z (commit-id=04c5116c9bdf8b762acc54b5500a9a276a5ec05a)
Runtime: go1.22.7 linux/amd64
Copyright (c) 2015-2024 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

The operator has deployed the Tenant with this Image:
`minio/minio:RELEASE.2024-06-13T22-53-53Z`

The operator itself:
`quay.io/minio/operator:v5.0.14`

 Is there an idea what the problem could be here?

**Comment by vadmeste** [comment]
Comparing wc of `mc tree` and `mc ls` is not correct, do you have a particular object that is shown `mc tree` and not in `mc ls --recursive`, can you run `mc stat` in that object ?
